torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2827,11 +2876,30 @@ def get_symbol(tensor_dim_src):
                 concrete_val = self.evaluate_expr(sympy.Eq(s1, s2))
                 if not concrete_val:
                     raise ConstraintViolationError(
-                        f"{src1.name()} = {self.var_to_val[s1]}"
+                        f"{src1.name()} = {s1.subs(self.var_to_val)}"
                         " is not equal to "
-                        f"{src2.name()} = {self.var_to_val[s2]}"
+                        f"{src2.name()} = {s2.subs(self.var_to_val)}"
                     )
 
+            for src, root, fn in equalities_inputs.derived_equalities:
+                s1 = get_symbol(src)
+                s2 = root if isinstance(root, sympy.Symbol) else get_symbol(root)
+                s2_ = fn(s2)
+                concrete_val = self.evaluate_expr(sympy.Eq(s1, s2_))
+                if not concrete_val:
+                    # we were unable to prove that src = fn(root) holds
+                    debug_name = self.var_to_sources[root][0].name() if isinstance(root, sympy.Symbol) else self._debug_name(root)
```

Comment:
It's weird to me that you can't just unconditionally call _debug_name here

Reply:
This disjoint union of cases appears in a few places (you'll see a few types in this PR being Union[this, that]), almost missing something like an algebraic datatype here. Or maybe it suggests that the code could be written differently.

Reply:
Moved around the logic here to make the case analysis clearer.